freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -20,13 +20,112 @@
 # SOFTWARE.
 #
 # Written by Valerij Maljulin <vmaljuli@redhat.com>
+# Written by Chuang Zhang <chuazhan@redhat.com>
 
-from freshmaker.handlers import ContainerBuildHandler
+import requests
+from http import HTTPStatus
+from requests.adapters import HTTPAdapter
+
+from freshmaker import conf, db, log
+from freshmaker.errata import Errata
 from freshmaker.events import FlatpakModuleAdvisoryReadyEvent
+from freshmaker.handlers import ContainerBuildHandler, fail_event_on_handler_exception
+from freshmaker.kojiservice import koji_service
+from freshmaker.lightblue import LightBlue
+from freshmaker.models import Event
+from freshmaker.pyxis import Pyxis
+from freshmaker.types import EventState
 
 
 class RebuildFlatpakApplicationOnModuleReady(ContainerBuildHandler):
     name = 'RebuildFlatpakApplicationOnModuleReady'
 
     def can_handle(self, event):
         return isinstance(event, FlatpakModuleAdvisoryReadyEvent)
+
+    @fail_event_on_handler_exception
+    def handle(self, event):
+
+        if event.dry_run:
+            self.force_dry_run()
+
+        self.event = event
+
+        db_event = Event.get_or_create_from_event(db.session, event)
+        self.set_context(db_event)
+
+        auto_build_images_list = self._get_auto_rebuild_image_list(event)
+        if not auto_build_images_list:
+            msg = ("There is no auto rebuild image can be rebuilt. "
+                   f"message_id: {event.msg_id}")
+            db_event.transition(EventState.SKIPPED, msg)
+            db.session.commit()
+            self.log_info(msg)
+            return []
+
+        rebuild_images = self._filter_images_with_higher_rpm_nvr(event, auto_build_images_list)
+        if not rebuild_images:
```

Comment:
Optional: there's some repeated code in this block and the previous one (about transitioning the event to SKIPPED), you might consider re-arranging the code so that you don't repeat it.

Reply:
Ack. Fixing